src/fides/api/ops/service/connectors/consent_email_connector.py

```diff
@@ -119,15 +135,36 @@ def batch_email_send(self, privacy_requests: Query) -> None:
             filtered_user_identities: Dict[
                 str, Any
             ] = filter_user_identities_for_connector(self.config, user_identities)
-            if filtered_user_identities and privacy_request.consent_preferences:
+            consent_preferences: List[PrivacyRequestConsentPreference] = [
+                PrivacyRequestConsentPreference(**pref)
+                for pref in privacy_request.consent_preferences or []
+            ]
+            filtered_consent_preferences: List[
+                PrivacyRequestConsentPreference
+            ] = filter_consent_preferences_for_propagation(
+                self.configuration.system, consent_preferences
+            )
+            if filtered_user_identities and filtered_consent_preferences:
                 batched_consent_preferences.append(
                     ConsentPreferencesByUser(
                         identities=filtered_user_identities,
-                        consent_preferences=privacy_request.consent_preferences,
+                        consent_preferences=filtered_consent_preferences,
                     )
                 )
             else:
                 skipped_privacy_requests.append(privacy_request.id)
+                ExecutionLog.create(
+                    db=db,
+                    data={
+                        "connection_key": self.configuration.key,
+                        "dataset_name": self.configuration.name,
+                        "collection_name": self.configuration.name,
+                        "privacy_request_id": privacy_request.id,
+                        "action_type": ActionType.consent,
+                        "status": ExecutionLogStatus.skipped,
+                        "message": f"Consent email skipped for '{self.configuration.name}'",
+                    },
+                )
```

Comment:
Creating skipped logs for consent email connectors if there ends up being no email to send. 

(Note that the email connectors actually batch multiple preferences from multiple users and are sent once weekly, so I'm going through all the relevant privacy requests for the weekly send here, and adding a skipped execution log if applicable)

This is for reporting that's coming soon, I want to make it easier to distinguish which systems were affected, or which systems we attempted to communicate with.